File: controller/src/main/java/yeetcd/controller/pipeline/WorkOutputPath.java
package yeetcd.controller.pipeline;

import yeetcd.protocol.pipeline.PipelineOuterClass;

public record WorkOutputPath(String name, String path) {

    public static WorkOutputPath fromProtobuf(PipelineOuterClass.WorkOutputPath workOutputPath) {
        return new WorkOutputPath(workOutputPath.getName(), workOutputPath.getPath());
    }
}

File: golang/internal/core/pipeline/work_output_path.go
package pipeline

// WorkOutputPath represents an output path for work
type WorkOutputPath struct {
	Name string
	Path string
}

// FromProtobuf converts protobuf WorkOutputPath to Go struct
func WorkOutputPathFromProtobuf(protoWorkOutputPath interface{}) (*WorkOutputPath, error) {
	return nil, nil
}
